source/base/data_out_base.cc

```diff
@@ -5597,43 +5622,80 @@ namespace DataOutBase
             out << data_names[last_component];
           }
 
-        out << "\" NumberOfComponents=\"3\" format=\"" << ascii_or_binary
-            << "\">\n";
+        out << "\" NumberOfComponents=\"" << (is_tensor ? "9" : "3")
```

Comment:
Please create a variable `n_components` and store the appropriate value in there.